lib/server-info-route-handler.js

```diff
@@ -0,0 +1,150 @@
+const _ = require('lodash')
+const { calculateBounds } = require('@terraformer/spatial')
+const {
+  getCollectionCrs,
+  getGeometryTypeFromGeojson,
+  normalizeExtent,
+  normalizeSpatialReference
+} = require('./helpers')
+const { serverMetadata: serverMetadataDefaults } = require('./defaults')
+const debug = process.env.KOOP_LOG_LEVEL === 'debug' || process.env.LOG_LEVEL === 'debug'
+
+function serverMetadata (json, { query = {} } = {}) {
+  const { extent, metadata, ...rest } = json
+  const { maxRecordCount, hasStaticData, description } = { ...metadata, ...rest }
+  const spatialReference = getSpatialReference(json, query)
+  const { layers, tables } = normalizeInput(json)
+  const fullExtent = getServiceExtent({ extent, metadata, layers, spatialReference })
+
+  // TODO reproject default extents when non WGS84 CRS is found or passed
+
+  return _.defaults({
+    spatialReference,
+    fullExtent,
+    initialExtent: fullExtent,
+    layers: layers.map(layerInfo),
+    tables: tables.map(tableInfo),
+    serviceDescription: description,
+    maxRecordCount: maxRecordCount || _.get(layers, '[0].metadata.maxRecordCount'),
+    hasStaticData: typeof hasStaticData === 'boolean' ? hasStaticData : false
+  }, serverMetadataDefaults)
+}
+
+function normalizeInput (input) {
+  const { type, tables = [], layers = [] } = input
+
+  if (type === 'FeatureCollection') {
+    const geometryType = getGeometryTypeFromGeojson(input)
+    if (geometryType) return { layers: [input], tables }
+    return { tables: [input], layers }
+  }
+
+  // To ensure tables and layers have been properly segregated merge and re-bin
+  const merged = _.concat(layers, tables)
+
+  return categorizeTableAndLayers(merged)
```

Comment:
Why do you combine the array and re-do the categorization?

Reply:
Thanks for asking. There is no guarantee that the incoming `layers` and `tables` have been properly segregated (anything without a geometryType should be a table).  This acts as a correction for uniformed users that pass all collections in `layers` even if the data has no geometry.  In fact, we had an integration test that passed in a tabular collection in the `layers` array, with the resulting response being wrong (empty `tables: []`).

I'm chose to re-do categorization in an effort to simplify `getData` operations, but we could instead just do a validation an either reject or warn when users wrongly put tabular data into the `layers` input array.